app/api/runQuery/route.ts

```diff
@@ -0,0 +1,26 @@
+import { NextResponse } from "next/server";
+import { kv } from "@vercel/kv";
+import { makeApiRequestRefreshingToken } from "@/shared/ApiHandlers/makeApiRequestRefreshingToken";
+import { SalesforceAuthCache } from "@/shared/types/salesforceTypes";
+
+export async function POST(req: Request) {
```

Comment:
Also confused why you have this endpoint alongside the `chat` endpoint. security issue here as well